src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/config/AuditConfig.java

```diff
@@ -1,45 +1,130 @@
 package com.amazon.opendistroforelasticsearch.security.auditlog.config;
 
 import com.amazon.opendistroforelasticsearch.security.auditlog.impl.AuditCategory;
+import com.amazon.opendistroforelasticsearch.security.compliance.ComplianceConfig;
 import com.amazon.opendistroforelasticsearch.security.support.ConfigConstants;
 import com.amazon.opendistroforelasticsearch.security.support.WildcardMatcher;
 
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
 import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
 import com.google.common.collect.ImmutableSet;
 import org.apache.logging.log4j.Logger;
 import org.elasticsearch.common.settings.Settings;
 
 import java.util.Collections;
 import java.util.EnumSet;
 import java.util.List;
+import java.util.Map;
 import java.util.Set;
 
+import static com.amazon.opendistroforelasticsearch.security.DefaultObjectMapper.getOrDefault;
+
 /**
  * Class represents configuration for audit logging.
+ * Expected class structure
+ * {
+ *   "enabled": true,
+ *   "audit" : {
+ *     "enable_rest" : true,
+ *     "disabled_rest_categories" : [
+ *       "GRANTED_PRIVILEGES",
+ *       "SSL_EXCEPTION"
+ *     ],
+ *     "enable_transport" : true,
+ *     "disabled_transport_categories" : [
+ *       "GRANTED_PRIVILEGES",
+ *       "AUTHENTICATED"
+ *     ],
+ *     "resolve_bulk_requests" : false,
+ *     "log_request_body" : true,
+ *     "resolve_indices" : true,
+ *     "exclude_sensitive_headers" : true,
+ *     "ignore_users" : [
+ *       "kibanaserver"
+ *     ],
+ *     "ignore_requests" : [ ]
+ *   },
+ *   "compliance" : {
+ *     "enabled": true,
+ *     "internal_config" : true,
+ *     "external_config" : true,
+ *     "read_metadata_only" : true,
+ *     "read_watched_fields" : { },
+ *     "read_ignore_users" : [ ],
+ *     "write_metadata_only" : true,
+ *     "write_log_diffs" : false,
+ *     "write_watched_indices" : [ ],
+ *     "write_ignore_users" : [ ]
+ *   }
+ * }
  */
 public class AuditConfig {
 
     public static final List<String> DEFAULT_IGNORED_USERS = Collections.singletonList("kibanaserver");
 
-    private AuditConfig() { }
+    private AuditConfig() {
+        this(true, null, null);
+    }
+
+    @JsonProperty("enabled")
+    private final boolean auditLogEnabled;
+    @JsonProperty("audit")
+    private final Filter filter;
+    @JsonProperty("compliance")
```

Comment:
don't think you need @JsonProperty here since the variable name is compliance

Reply:
By default only public properties are auto detected by Jackson (if I remember correctly).

Reply:
It seems to work because the property/getter with visibility off and variable name matches.